src/Sulu/Bundle/PageBundle/Resources/js/views/PageList/PageList.js

```diff
@@ -53,20 +54,28 @@ class PageList extends React.Component<Props> {
             return;
         }
 
-        if (webspace.allLocalizations.find((localization) => localization.localization === userStore.contentLocale)) {
-            this.locale.set(userStore.contentLocale);
+        let locale = null;
 
-            return;
+        if (webspace.allLocalizations.find((localization) => localization.localization === userStore.contentLocale)) {
+            locale = userStore.contentLocale;
         }
 
-        const locale = this.findDefaultLocale(webspace.localizations);
+        if (!locale) {
+            locale = this.findDefaultLocale(webspace.localizations);
+        }
 
         if (!locale) {
             throw new Error(
                 'Default locale in webspace "' + webspace.key + '" not found'
             );
         }
 
+        if (locale === this.locale.get()) {
+            return;
+        }
+
+        router.redirect(router.route.name, {...router.attributes, locale: locale});
+
         this.locale.set(locale);
```

Comment:
This should not be necessary anymore, because the redirect before will change this value already.